styleguide/source/assets/scss/01-atoms/_table.scss

```diff
@@ -18,82 +18,96 @@ table {
   }
 }
 
-tbody {
+caption {
   display: block;
 
-  tr {
-    display: block;
+  @include breakpoint($bp-small) {
+    display: table-caption;
   }
+}
 
-  tr.odd, tr:nth-child(odd)  {
-    background: $gray-7;
+thead {
+  display: block;
+
+  @include breakpoint($bp-small) {
+    display: table-header-group;
   }
 
-  td {
-    border: 1px solid $gray-7;
-    display: block;
-    padding-left: 50%;
-    position: relative;
+  tr {
+    text-transform: uppercase;
+    position: absolute;
+    top: -9999px;
+    left: -9999px;
+
     @include breakpoint($bp-small) {
-      padding-left: 0;
+      position: static;
+    }
+
+    th {
+      display: block;
       border: 1px solid $gray-50;
+
+      @include breakpoint($bp-small) {
+        display: table-cell;
+      }
     }
   }
+}
 
-  td:before {
-    font-weight: bold;
-    text-transform: uppercase;
-    content: attr(data-title);
-    position: absolute;
-    top: 6px;
-    left: 6px;
-    width: 45%;
-    white-space: normal;
-  }
+tbody {
+  display: block;
 
   @include breakpoint($bp-small) {
     display: table-row-group;
+  }
 
-    tr {
+  tr {
+    display: block;
+
+    @include breakpoint($bp-small) {
       display: table-row;
     }
 
-    td {
-      display: table-cell;
-      vertical-align: top;
+    &.odd,
+    &:nth-child(odd)  {
+      background: $gray-7;
     }
 
-    td:before {
-      display: none;
+    td {
+      border: 1px solid $gray-7;
+      display: block;
+      padding: 2px 2px 2px 50%;
+      position: relative;
+
+      @include breakpoint($bp-small) {
+        display: table-cell;
+        vertical-align: top;
+        padding: 2px;
+        border: 1px solid $gray-50;
+      }
+
+      &:before {
+        font-weight: bold;
+        text-transform: uppercase;
+        content: attr(data-title);
```

Comment:
This is correct, but I want to double check how this is expecting the `data-title` to be set. We need to build this in such a way that it just expect normal tabular information (just `th` and `td`) and it is on us to ensure that the code dynamically pulls the correct `th` to place as the `data-title` on the appropriate `td`.

Reply:
@labbydev -- If I understand correctly, on the Drupal end we'll need a simple `template_preprocess_table(&$variables)` to add `data-title` to each table.